src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -426,25 +427,26 @@ private IQueryCache BuildQueryCache(string queryCacheName)
 					properties);
 		}
 
-		private ICacheConcurrencyStrategy GetCacheConcurrencyStrategy(
+		private Func<string, ICacheConcurrencyStrategy> GetCacheConcurrencyStrategy(
```

Comment:
Instead of cache instance returns delegate that will be used by persisters to obtain cache instance by `tenantIdentifier`.